identity/scim.go

```diff
@@ -116,21 +121,52 @@ type GroupList struct {
 	Resources    []ScimGroup `json:"resources,omitempty"`
 }
 
-// GroupPatchRequest contains a request structure to make a patch op against SCIM api
-type GroupPatchRequest struct {
-	Schemas    []URN                  `json:"schemas,omitempty"`
-	Operations []GroupPatchOperations `json:"Operations,omitempty"`
+var entitlementMapping = map[string]string{
+	"allow-cluster-create":       "allow_cluster_create",
+	"allow-instance-pool-create": "allow_instance_pool_create",
+	"databricks-sql-access":      "allow_sql_analytics_access", // TODO: state change
 }
 
-// Entitlement is a custom type that contains a set of entitlements for a user/group
-type Entitlement string
+// order is important for tests
+var possibleEntitlements = []string{
+	"allow-cluster-create",
+	"allow-instance-pool-create",
+	"databricks-sql-access"}
```

Comment:
maybe generate it from `entitlementMapping`, like: `reflect.ValueOf(entitlementMapping).MapKeys()` - it could be easier to maintain if we add more entitlements